Step2/Install_Win_Features.ps1

```diff
@@ -0,0 +1,86 @@
+# Import the configuration
+Import-Module -Name .\Config.ps1
+
+# Function to handle logging and console output
+function Log-Message {
+    param (
+        [string]$Message
+    )
+    
+    $Timestamp = (Get-Date).ToString("yyyy-MM-dd HH:mm:ss")
+    "$Timestamp - $Message" | Out-File -FilePath $Global:LogFilePath -Append
+    Write-Host $Message
+}
+
+function Install_Win_Features {
+    param (
+        [string[]]$Features
+    )
+    
+    $InstallationSuccessful = $true
+
+    foreach ($Feature in $Features) {

```

Comment:
The above comment could lead to ambiguous inputs being passed to  your function and a failure of this foreach loop.

How can you ensure that this does not happen?